src/uct/tcp/sockcm/sockcm_iface.c

```diff
@@ -94,29 +124,165 @@ static uct_iface_ops_t uct_sockcm_iface_ops = {
 
 void uct_sockcm_iface_client_start_next_ep(uct_sockcm_iface_t *iface)
 {
-    ucs_trace("sockcm_iface_client_start_next_ep not implemented");
+    ucs_status_t status;
+    uct_sockcm_ep_t *ep, *tmp;
+
+    UCS_ASYNC_BLOCK(iface->super.worker->async);
+
+    /* try to start an ep from the pending eps list */
+    ucs_list_for_each_safe(ep, tmp, &iface->pending_eps_list, list_elem) {
+        status = uct_sockcm_ep_set_sock_id(iface, ep);
+        if (status != UCS_OK) {
+            continue;
```

Comment:
why need to continue?

Reply:
Refer https://github.com/openucx/ucx/pull/4073#discussion_r315900884